web/src/platform/singlethreaded/wasm_entries.rs

```diff
@@ -1,95 +1,89 @@
-use std::{mem, rc::Rc};
+use std::{
+    fmt::{Display, Formatter},
+    mem,
+    rc::Rc,
+};
 
-use js_sys::{ArrayBuffer, Uint8Array};
-use log::{error, info};
+use js_sys::ArrayBuffer;
 use maplibre::{
     benchmarking::io::{
-        apc::{AsyncProcedure, Input, Message},
+        apc::{AsyncProcedure, Input},
         source_client::{HttpSourceClient, SourceClient},
     },
-    io::transferables::Transferables,
+    io::apc::CallError,
 };
 use wasm_bindgen::{prelude::*, JsCast};
 
 use crate::{
+    error::WrappedError,
     platform::singlethreaded::{
-        apc::{ReceivedType, SerializedMessageTag},
+        apc::{MessageTag, ReceivedType},
         transferables::FlatBufferTransferable,
-        PassingContext, UsedContext, UsedTransferables,
+        PassingContext, UsedContext,
     },
     WHATWGFetchHttpClient,
 };
 
 /// Entry point invoked by the worker.
 #[wasm_bindgen]
-pub async fn singlethreaded_worker_entry(procedure_ptr: u32, input: String) -> Result<(), JsValue> {
+pub async fn singlethreaded_worker_entry(
+    procedure_ptr: u32,
+    input: String,
+) -> Result<(), WrappedError> {
     let procedure: AsyncProcedure<UsedContext> = unsafe { mem::transmute(procedure_ptr) };
 
-    let input = serde_json::from_str::<Input>(&input).unwrap(); // FIXME (wasm-executor): Remove unwrap
+    let input =
+        serde_json::from_str::<Input>(&input).map_err(|e| CallError::Deserialize(Box::new(e)))?;
 
     let context = PassingContext {
         source_client: SourceClient::new(HttpSourceClient::new(WHATWGFetchHttpClient::new())),
     };
 
-    let result = (procedure)(input, context).await;
-
-    if let Err(e) = result {
-        error!("{:?}", e); // TODO handle better
-    }
+    (procedure)(input, context).await?;
```

Comment:
just for personal education - why do you need to put a var in the parenthesis here?

Reply:
They are probably not required here. Though there are cases where you need them. I think if `procedure` is part of a struct, then you have to do `(structure.procedure)()` because you are calling the lambda and not a function defined on the struct.

A new opportunity for a clippy lint? :D